File: python/basic_tips/inheritance.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-


class Parent:
    def __init__(self, x):
        self.x = x
        self._prop = x

    @property
    def prop(self):
        return self._prop

    @prop.setter
    def prop(self, v):
        self._prop = v


class Child(Parent):
    prop = property(Parent.prop.__get__)

    def __init__(self, x):
        super().__init__(x)
        print('i aaaaaaaaam a chiiiiiiiiiiiild')

    @prop.setter
    def prop(self, v):
        print(f'prop has set as {v}')
        self._prop = v


c = Child(10)
c.prop = 20

File: javascript/basic_tips/inheritance.js
class Parent {
  constructor(value) {
    this.value = value
  }

  instance_method() {
    console.log('I am parent')
    console.log(this.value)
  }
}

class Child extends Parent {
  instance_method() {
    super.instance_method()
    console.log('I am child')
    console.log(this.value)
  }
}

var child = new Child(10)
child.instance_method()
